squirrel.deduplication/src/main/java/org/dice_research/squirrel/sink/impl/sparql/QueryGenerator.java

```diff
@@ -0,0 +1,234 @@
+        stringBuilder.append(" ");
+        stringBuilder.append(hashValue);
+        stringBuilder.append("}}");
+        LOGGER.info("Dedup_Testing: query getHashQuery: " + stringBuilder.toString());
+        Query query = QueryFactory.create(stringBuilder.toString());
+        return query;
+    }
+    /**
+     * Return a select query for the given graphID or default graph.
+     * It will return all triples contained in the graph.
+     *
+     * @param graphID      The id of the graph from which you want to select.
+     * @param defaultGraph Identify if query is for the default graph
+     * @return All triples contained in the graph.
+     */
+    public Query getSelectQuery(String graphID, boolean defaultGraph) {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append(getPrefixes());
+        stringBuilder.append("SELECT ?subject ?predicate ?object WHERE { ");
+        if (!defaultGraph) {
+            stringBuilder.append("GRAPH <");
+            stringBuilder.append(graphID);
+            stringBuilder.append("> { ");
+        }
+        stringBuilder.append("?subject ?predicate ?object ");
+        if (!defaultGraph) {
+            stringBuilder.append("} ");
+        }
+        stringBuilder.append("}");
+        LOGGER.info("Dedup_Testing: query getSelectQuery: " + stringBuilder.toString());
+        Query query = QueryFactory.create(stringBuilder.toString());
+        return query;
+    }
+
+    public Query getDeleteQuery(String graphID) {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append(getPrefixes());
+        stringBuilder.append("DELETE WHERE {");
+        stringBuilder.append("GRAPH <");
+        stringBuilder.append(graphID);
+        stringBuilder.append("> { ");
+        stringBuilder.append("?subject ?predicate ?object ");
+        stringBuilder.append("}}");
+        LOGGER.info("Dedup_Testing: query getDeleteQuery: " + stringBuilder.toString());
+        Query query = QueryFactory.create(stringBuilder.toString());
+        return query;
+    }
+
+    public Query getUpdateTriplesGraphIdQuery(String newUri, RDFNode oldGraphId) {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append(getPrefixes());
+        stringBuilder.append("DELETE { GRAPH <");
+        stringBuilder.append(METADATA_GRAPH_ID);
+        stringBuilder.append("> { ?subject sq:containsDataOf <");
+        stringBuilder.append(newUri);
+        stringBuilder.append("> }}");
+        stringBuilder.append("INSERT { GRAPH <");
+        stringBuilder.append(METADATA_GRAPH_ID);
+        stringBuilder.append("> {");
+        stringBuilder.append(oldGraphId.toString());
+		stringBuilder.append(" sq:containsDataOf <");
+		stringBuilder.append(newUri);
+		stringBuilder.append("> } }");
+		stringBuilder.append(" WHERE { GRAPH <");
+		stringBuilder.append(METADATA_GRAPH_ID);
+		stringBuilder.append("> { ?subject ?object ?predicate } }");
```

Comment:
This query seems to be buggy. I may do what you want it to do but it has the ability to delete much more information.